config/externalname.go

```diff
@@ -430,6 +430,8 @@ var terraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
 	"google_dataplex_asset": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/locations/{{ .parameters.location }}/lakes/{{ .parameters.lake }}/zones/{{ .parameters.dataplex_zone }}/assets/{{ .external_name }}"),
 	// Imported by using the following projects/{{project}}/locations/{{location}}/lakes/{{lake}}/zones/{{name}}
 	"google_dataplex_zone": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/locations/{{ .parameters.location }}/lakes/{{ .parameters.lake }}/zones/{{ .external_name }}"),
+	// Imported by using the following projects/{{project}}/locations/{{location}}/aspectTypes/{{aspect_type_id}}
+	"google_dataplex_aspect_type": config.IdentifierFromProvider,
```

Comment:
For external name normalization, could you please try the following [TemplatedStringAsIdentifier](https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#gcp) configuration? If it works, please mark the location parameter as required as shown [here](https://github.com/crossplane-contrib/provider-upjet-gcp/blob/743ed2c297b8d3738c18f76b00382ee2252a82cb/config/cloudrun/config.go#L43).

```suggestion
	"google_dataplex_aspect_type": TemplatedStringAsIdentifier("aspect_type_id", "projects/{{ .setup.configuration.project }}/locations/{{ .parameters.location }}/aspectTypes/{{ .external_name }}"),
```

Reply:
thanks for the feedback @turkenf, i had initially tried this but i've noticed the wrong external_name was being used. The correct change has been made inline with your comments.

Reply:
Thank you :)